easybuild/easyblocks/j/java.py

```diff
@@ -76,15 +78,128 @@ def extract_step(self):
+
+                for ld_glob_pattern in (r'ld-linux-*.so.*', r'ld*.so.*'):
+                    res = glob.glob(os.path.join(sysroot, 'lib*', ld_glob_pattern))
+                    self.log.debug("Paths for ELF interpreter via '%s' pattern: %s", ld_glob_pattern, res)
+                    if res:
+                        # if there are multiple hits, make sure they resolve to the same paths,
+                        # but keep using the symbolic link, not the resolved path!
+                        real_paths = nub([os.path.realpath(x) for x in res])
+                        if len(real_paths) == 1:
+                            elf_interp = res[0]
+                            self.log.info("ELF interpreter found at %s", elf_interp)
+                            break
+                        else:
+                            raise EasyBuildError("Multiple different unique ELF interpreters found: %s", real_paths)
+
+                if elf_interp is None:
+                    raise EasyBuildError("Failed to isolate ELF interpreter!")
+
+                bindir = os.path.join(self.installdir, 'bin')
+                for path in os.listdir(bindir):
+                    path = os.path.join(bindir, path)
+                    out, _ = run_cmd("file %s" % path, trace=False)
+                    if "dynamically linked" in out:
+
+                        out, _ = run_cmd("patchelf --print-interpreter %s" % path, trace=False)
+                        self.log.debug("ELF interpreter for %s: %s" % (path, out))
+
+                        run_cmd("patchelf --set-interpreter %s %s" % (elf_interp, path), trace=False)
+
+                        out, _ = run_cmd("patchelf --print-interpreter %s" % path, trace=False)
+                        self.log.debug("ELF interpreter for %s: %s" % (path, out))
+
+                        out, _ = run_cmd("patchelf --print-rpath %s" % path, simple=False, trace=False)
+                        curr_rpath = out.strip()
+                        self.log.debug("RPATH for %s: %s" % (path, curr_rpath))
+
+                        new_rpath = ':'.join([curr_rpath] + sysroot_lib_paths)
+                        # note: it's important to wrap the new RPATH value in single quotes,
+                        # to avoid magic values like $ORIGIN being resolved by the shell
+                        run_cmd("patchelf --set-rpath '%s' %s" % (new_rpath, path), trace=False)
+
+                        curr_rpath, _ = run_cmd("patchelf --print-rpath %s" % path, simple=False, trace=False)
+                        self.log.debug("RPATH for %s (prior to shrinking): %s" % (path, curr_rpath))
+
+                        run_cmd("patchelf --shrink-rpath %s" % path, trace=False)
+
+                        curr_rpath, _ = run_cmd("patchelf --print-rpath %s" % path, simple=False, trace=False)
+                        self.log.debug("RPATH for %s (after shrinking): %s" % (path, curr_rpath))
+
+                libdir = os.path.join(self.installdir, 'lib')
```

Comment:
Same here, can we just check all the paths that are being added to `LIBRARY_PATH`